src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -55,85 +64,70 @@ const defaultProps = {
     bankAccountList: [],
     cardList: [],
     isLoadingPayments: false,
+    enableSelection: false,
+    selectedAccountID: '',
+    filterList: '',
     isAddPaymentMenuActive: false,
 };
 
 class PaymentMethodList extends Component {
     constructor(props) {
         super(props);
-
+        this.state = {
+            selectedAccountID: this.props.selectedAccountID,
+        };
         this.renderItem = this.renderItem.bind(this);
+        this.selectPaymentMethod = this.selectPaymentMethod.bind(this);
+        this.createPaymentMethodList = this.createPaymentMethodList.bind(this);
     }
 
     /**
      * Take all of the different payment methods and create a list that can be easily digested by renderItem
-     *
-     * @returns {Array}
+     * @param {String} filter
+     * @return {Array}
      */
-    createPaymentMethodList() {
-        const combinedPaymentMethods = [];
-
-        _.each(this.props.bankAccountList, (bankAccount) => {
-            // Add all bank accounts besides the wallet
-            if (bankAccount.type !== CONST.BANK_ACCOUNT_TYPES.WALLET) {
-                const formattedBankAccountNumber = bankAccount.accountNumber
-                    ? `${this.props.translate('paymentMethodList.accountLastFour')} ${
-                        bankAccount.accountNumber.slice(-4)
-                    }`
-                    : null;
-                const {icon, iconSize} = getBankIcon(lodashGet(bankAccount, 'additionalData.bankName', ''));
-                combinedPaymentMethods.push({
-                    type: MENU_ITEM,
-                    title: bankAccount.addressName,
-
-                    // eslint-disable-next-line
-                    description: formattedBankAccountNumber,
-                    icon,
-                    iconSize,
-                    onPress: e => this.props.onPress(e, bankAccount.bankAccountID),
-                    key: `bankAccount-${bankAccount.bankAccountID}`,
-                });
+    createPaymentMethodList(filter) {
+        const paymentMethods = getPaymentMethodsList(
+            !filter || filter === CONST.WALLET.PAYMENT_METHOD_TYPE.BANK ? this.props.bankAccountList : [],
```

Comment:
This feels really messy to me. Can you think of any other way we could filter these out with the same effect? I'm drawing a blank..